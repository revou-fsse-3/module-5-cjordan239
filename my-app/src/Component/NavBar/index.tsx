
import Link from 'next/link';
import styles from './styles/style-navbar.module.css';
import { useRouter } from 'next/router';

const NavBar = () => {
  return (
    <nav className={styles['Navbar']}>
      <h1> NewsAPP </h1>
      <div>
        <ul className={styles.NavbarMenu}>
          <li>
            <Link href='/home'>
              Home
            </Link>
          </li>
          <li>
            <Link href='/register'>
              Register
            </Link>
          </li>
          <li>
            <Link href='/login'>
              Login
            </Link>
          </li>
        </ul>
      </div>
    </nav>
  );
};

export default NavBar;
