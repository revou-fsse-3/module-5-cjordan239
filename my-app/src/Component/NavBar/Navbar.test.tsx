import { render, screen } from '@testing-library/react';
import NavBar from '.';
import React from 'react';

describe('Unit test for NavBar', () => {
    it('should render the navbar with login', () => {
        render(<NavBar />);
        expect(screen.getByText('Login')).toBeDefined();
        expect(screen.getByText('Login')).toMatchSnapshot();
    });

    test('should render navbar with home ', () => {
        render(<NavBar />);
        expect(screen.getByText('Login')).toBeDefined();
        expect(screen.getByText('Login')).toMatchSnapshot();
    });
});
