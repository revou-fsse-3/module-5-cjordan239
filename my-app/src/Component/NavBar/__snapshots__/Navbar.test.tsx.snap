// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit test for NavBar should render navbar with home  1`] = `
<a
  href="/login"
>
  Login
</a>
`;

exports[`Unit test for NavBar should render the navbar with login 1`] = `
<a
  href="/login"
>
  Login
</a>
`;
